gr-blocks/python/blocks/parse_file_metadata.py

```diff
@@ -75,12 +76,21 @@ def parse_header(p, VERBOSE=False):
         r = pmt.dict_ref(p, pmt.string_to_symbol("rx_time"), dump)
         secs = pmt.tuple_ref(r, 0)
         fracs = pmt.tuple_ref(r, 1)
-        secs = float(pmt.to_uint64(secs))
+        secs = pmt.to_uint64(secs)
         fracs = pmt.to_double(fracs)
-        t = secs + fracs
+        t = float(secs) + fracs
+        info["rx_time_secs"] = secs
+        info["rx_time_fracs"] = fracs
         info["rx_time"] = t
         if(VERBOSE):
-            print("Seconds: {0:.6f}".format(t))
+            with decimal.localcontext() as ctx:
+                # we need 36 decimal places in order not to lose precision
+                # since an uint64 can contain 20 decimal places and the
+                # resolution we should expect from the fracs duoble
+                # is the machine epsilon, which is ~1e-16
+                ctx.prec = 36
+                s = decimal.Decimal(secs) + decimal.Decimal(fracs)
+                print("Seconds: {0}.{1}".format(s))
```

Comment:
This needs to be updated. You could update the code like so:
```suggestion
                print("Seconds: {0:.16f}".format(s))
```

Reply:
This is done in 0dfb8c794c01f481556afe952b7904e96f8d4e39